homeassistant/components/smartthings/climate.py

```diff
@@ -254,5 +303,128 @@ def target_temperature_low(self):
+            FAN_MODE_TO_SPEED[fan_mode], set_status=True)
+        # State is set optimistically in the command above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state()
+
+    async def async_set_operation_mode(self, operation_mode):
+        """Set new target operation mode."""
+        await self._device.set_air_conditioner_mode(
+            STATE_TO_AC_MODE[operation_mode], set_status=True)
+        # State is set optimistically in the command above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state()
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        tasks = []
+        # operation mode
+        operation_mode = kwargs.get(ATTR_OPERATION_MODE)
+        if operation_mode:
+            tasks.append(self.async_set_operation_mode(operation_mode))
+        # temperature
+        tasks.append(self._device.set_cooling_setpoint(
+            kwargs[ATTR_TEMPERATURE], set_status=True))
+        await asyncio.gather(*tasks)
+        # State is set optimistically in the command above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state()
+
+    async def async_turn_on(self):
+        """Turn device on."""
+        await self._device.switch_on(set_status=True)
+        # State is set optimistically in the command above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state()
+
+    async def async_turn_off(self):
+        """Turn device off."""
+        await self._device.switch_off(set_status=True)
+        # State is set optimistically in the command above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state()
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        return SPEED_TO_FAN_MODE.get(self._device.status.fan_speed)
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return AC_MODE_TO_STATE.get(self._device.status.air_conditioner_mode)
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._device.status.temperature
+
+    @property
+    def device_state_attributes(self):
+        """
+        Return device specific state attributes.
+
+        Include attributes from the Demand Response Load Control (drlc)
+        and Power Consumption capabilities.
+        """
+        attributes = [
+            'drlc_status_duration',
+            'drlc_status_level',
+            'drlc_status_start',
+            'drlc_status_override',
+            'power_consumption_start',
```

Comment:
Is it absolute utc?

Reply:
SmartThings doesn't define it but based on the data samples from users, it looks to be UTC (i.e. `2019-02-24T21:03:04Z`).  This applies to `drlc_status_start`, `power_consumption_start`, and `power_consumption_end`.